src/server/qgis_map_serv.cpp

```diff
@@ -384,6 +386,28 @@ int main( int argc, char * argv[] )
 
     // Copy the parameters map
     QMap<QString, QString> parameterMap( theRequestHandler->parameterMap() );
+    QMap<QString, QString> parameters;
+    foreach ( QString element, QString( getenv( "QUERY_STRING" ) ).split( "&" ) )
+    {
+      int sepidx = element.indexOf( "=", 0, Qt::CaseSensitive );
+      if ( sepidx == -1 )
+      {
+        continue;
+      }
+
+      QString key = element.left( sepidx );
+      key = QUrl::fromPercentEncoding( key.toLocal8Bit() ); //replace encoded special characters and utf-8 encodings
+
+      QString value = element.mid( sepidx + 1 );
+      value.replace( "+", " " );
+      value = QUrl::fromPercentEncoding( value.toLocal8Bit() ); //replace encoded special characters and utf-8 encodings
+
+      parameters.insert( key, value );
+    }
+    QMap<QString, QString> headers;
+    // TODO
+    QgsRequest request = QgsRequest(&parameters, &headers);
+    QgsAccessControl accessControl = QgsAccessControl(&pluginsSecurity, &request);
```

Comment:
Why do you pass the "request"? The request handler already contains the parameter map through serverInterface